sky/task.py

```diff
@@ -862,8 +862,11 @@ def __repr__(self):
             s += f'\n  outputs: {self.outputs}'
         if self.num_nodes > 1:
             s += f'\n  nodes: {self.num_nodes}'
-        if len(self.resources) > 1 or not list(self.resources)[0].is_empty():
+        if len(self.resources) > 1:
             s += f'\n  resources: {self.resources}'
+        elif len(
+                self.resources) == 1 and not list(self.resources)[0].is_empty():
+            s += f'\n  resources: {list(self.resources)[0]}'
```

Comment:
Came across a bug (unrelated to this PR, but probably to easy to fix while we're at it):

```
import sky

t1 = sky.Task().set_resources({sky.Resources()})
t2 = sky.Task().set_resources({sky.Resources(), sky.Resources()})

print(f'T1 {t1}')
print(f'T2 {t2}')
```

Output:
```
T1 Task(run=None)
  resources: default instances
T2 Task(run=None)
  resources: {<Cloud>(), <Cloud>()}
```

Output I was expecting for T2:
```
T2 Task(run=None)
  resources: {<Cloud>(), <Cloud>()}
```

We should add a check for duplicated empty Resources and print them as a single 'default resources' if all `self.resources` are empty. 

Feel free to file as separate issue.

Reply:
This edge case will likely be removed in the future, if we drop the highly experimental and unused set-based API.